x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.ts

```diff
@@ -11,6 +11,12 @@ export interface CreatedPipelines {
   created: string[];
 }
 
+export interface PipelineDefinition {
+  processors?: Array<Record<string, any>>;
```

Comment:
suggestion: the Elasticsearch client already has a PipelineDefinition you can use. It lacks a version field, but if that's necessary you can extend it instead of building one from scratch. Giving it a more specific name would also help disambiguate type definitions. Something like:

```
export interface ESPipeLineDefinition extends PipelineDefinition {
  version?: number;
}
```

Reply:
Where is that type `PipelineDefinition` in `ElasticsearchClient`, where should I import from?
Can you advise if there's documentation on Kibana core types? I feel like I blindly search for things to copy-paste from codebase, and i can't always find them. :( This is a very sh*tty dev process.

Reply:
The documentation for the types is the types themselves with maybe comments inside, that's the convention across the TypeScript ecosystem. If you use VS Code or a different IDE that contain suggestions they'll help you out a lot. In general, you should be able to click through to type definitions from anything you're using in code (command-click in VS code) and letting the IDE/linter suggest the correct auto-import for you. The best dev process is generally to write the code with no or as few types as possible, and then let the linter/IDE guide you as to what is still missing.

In general, you can assume that whatever you haven't explicitly built in your PR is already typed and there is a type you can import if you need it explicitly.